src/platform/middleware.ts

```diff
@@ -73,3 +75,45 @@ export const gracefullyDenyDeletedDevices: RequestHandler = (
 			next();
 		});
 };
+
+// register the device state change handler which updates the device model...
+_.once(() => {
+	DeviceOnlineState.manager.on(
+		'change',
+		(uuid: string, newState: DeviceOnlineState.DeviceOnlineStates) => {
+			if (!uuid) return;
+
+			// patch the api_heartbeat_state value to the new state...
+			resinApi.patch({
+				resource: 'device',
+				options: {
+					$filter: {
+						uuid,
+						api_heartbeat_state: {
+							$ne: newState,
+						},
+					},
+				},
+				body: {
+					api_heartbeat_state: newState,
+				},
+				passthrough: {
+					req: root,
+				},
+			});
+		},
+	);
+})();
+
+export const registerDeviceContact = (path: string) => {
+	return ((req, _res, next) => {
+		const uuid = _.get(req, path, '');
+		if (uuid !== '') {
+			DeviceOnlineState.getPollIntervalForDevice(uuid).then(pollInterval =>
+				DeviceOnlineState.manager.captureEventFor(uuid, pollInterval / 1000),
+			);
+		}
+
+		next();
+	}) as RequestHandler;
+};
```

Comment:
All this code definitely doesn't belong here -- `platform` is much more generic than that and should know nothing about devices or their online state. In fact, we generally plan to move `platform` out of the API altogether into something more reusable.

Why having this code in `lib/device-online-state.ts` isn't good enough?

Reply:
@richbayliss was this marked as resolved by accident? I still think this does not belong here. Can you please address this or discuss if you think otherwise?